### ISSUE TITLE
How do we exclude `if` from the implicit_assignment_linter?

### ISSUE THREAD

**Issue by TimTaylor** [jimhester/lintr#2913]
The two attempts below fail to exclude `if`. Am I missing something obvious?

``` r
library(lintr)
lint(
    linters = list(implicit_assignment_linter(except = c("if", "`if`"))),
    text="if(a <- 1) 1"
)
#> <text>:1:4: warning: [implicit_assignment_linter] Avoid implicit assignments in function calls. For example, instead of `if (x <- 1L) { ... }`, write `x <- 1L; if (x) { ... }`.
#> if(a <- 1) 1
#>    ^~~~~~
```

<sup>Created on 2025-07-31 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
you might want allow_scoped=TRUE

**Comment by MichaelChirico** [comment]
if is an operator, not a function, I'm not sure it's worth mentioning.

**Comment by MichaelChirico** [comment]
That's right -- fixed in #2914.
